solutions/account-infrastructure-base/main.tf

```diff
@@ -19,4 +19,28 @@ module "account_infrastructure_base" {
   kms_key_crn                         = var.kms_key_crn
   resource_tags                       = var.resource_tags
   allowed_ip_addresses                = var.allowed_ip_addresses
+  custom_roles                        = var.custom_roles
+  access_groups = [{
```

Comment:
For the default, I'd suggest starting with a combination:
- 3 roles - admin, privileged, observer roles. Which respectively maps to admin.manager, editor/writer, viewer/reader permissions generally, but there are further mapping needed per service (eg: SecretReader for secret manager)
- group services per function - observability (atracker, logging, monitoring), security services (kms, hpcs, scc, , network (vpc), and compute (code engine, iks, roks, vsi, ...)

I would expose variables to allow end user to adjust:
- Define iam permissions per service for admin/privileged/observer roles.


eg: privileged_role_map with default:
```
{
  "default": [
    "Operator",
    "Writer"
  ],
  "kms": [
    "Operator",
    "Writer",
    "ReaderPlus"
  ],
  "logdna": [
    "Operator",
    "Manager"
  ],
  "logdnaat": [
    "Operator",
    "Manager"
  ],
  "secrets-manager": [
    "Operator",
    "Writer",
    "SecretsReader"
  ]
}
```

- Define list of services in each group
```
{
  "audit": {
    "description": "Audit and Observability Services",
    "services": [
      "logdnaat",
      "logdna"
    ]
  },
  "compute": {
    "description": "Compute services",
    "services": [
      "container-registry",
      "containers-kubernetes",
      "code-engine"
    ]
  },
  "security": {
    "description": "Security and Compliance related services",
    "services": [
      "secrets-manager",
      "kms",
       ...
    ]
  }
}
```

This gives a good balance in term of meeting least privilege req, without ending up with a large number of access groups. This also simplify the interface.

I would leave the access_groups variable as is but indicate that this is for advanced usage to target specific resource groups, create finer grained custom role, etc